scripts/update_config.js

```diff
@@ -21,119 +21,219 @@
 
 const fs = require('fs');
 const path = require('path');
+const readline = require('readline');
 const program = require('commander');
-const merge = require('lodash/merge');
+const lisk = require('lisk-elements').default;
 const observableDiff = require('deep-diff').observableDiff;
 const applyChange = require('deep-diff').applyChange;
+const JSONHistory = require('../helpers/json_history');
 
 const rootPath = path.resolve(path.dirname(__filename), '../');
 const loadJSONFile = filePath => JSON.parse(fs.readFileSync(filePath), 'utf8');
-let oldConfigPath;
-let newConfigPath;
+// Now get a unified config.json for 1.1.x version
+const defaultConfig = loadJSONFile(
+	path.resolve(rootPath, 'config/default/config.json')
+);
+let configFilePath;
+let fromVersion;
+let toVersion;
 
 program
 	.version('0.1.1')
-	.arguments('<old_config_file> <new_config_file>')
-	.action((oldConfig, newConfig) => {
-		oldConfigPath = oldConfig;
-		newConfigPath = newConfig;
+	.arguments('<input_file> <from_version> [to_version]')
+	.option('--output', 'Output file path')
+	.option('--diff', 'Show only difference from default config file.')
+	.action((inputFile, version1, version2) => {
+		fromVersion = version1;
+		toVersion = version2;
+		configFilePath = inputFile;
 	})
 	.parse(process.argv);
 
-if (!oldConfigPath || !newConfigPath) {
-	console.error('error: no config file provided.');
-	process.exit(1);
-}
-
-console.info('Running config migration from 1.0.x to 1.1.x');
-console.info('Starting configuration migration...');
-
-// Old config in 1.0.x will be single unified config file.
-const oldConfig = loadJSONFile(oldConfigPath);
-
-// Had dedicated ssl config only for API
-// https://github.com/LiskHQ/lisk/issues/2154
-if (oldConfig.ssl) {
-	oldConfig.api.ssl = merge({}, oldConfig.ssl);
-	delete oldConfig.ssl;
-}
-
-// New config in 1.1.x will be partial config other than default/config.json
-const newConfig = loadJSONFile(newConfigPath);
-
-// Now get a unified config.json for 1.1.x version
-const defaultConfig = loadJSONFile(
-	path.resolve(rootPath, 'config/default/config.json')
-);
-const unifiedNewConfig = merge({}, defaultConfig, newConfig);
+const history = new JSONHistory('lisk config file', console);
+
+history.version('0.9.x');
```

Comment:
Can we throw 'unsupported' error for the versions < 0.9.x?

Reply:
Done